src/Parsers/Access/ParserGrantQuery.cpp

```diff
@@ -284,8 +283,30 @@ bool ParserGrantQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     AccessRightsElements elements;
     std::shared_ptr<ASTRolesOrUsersSet> roles;
-    if (!parseElementsWithoutOptions(pos, expected, elements) && !parseRoles(pos, expected, is_revoke, attach_mode, roles))
-        return false;
+
+    bool current_grants = false;
+    if (!is_revoke && ParserKeyword{"CURRENT GRANTS"}.ignore(pos, expected))
+    {
+        current_grants = true;
+        if (ParserToken(TokenType::OpeningRoundBracket).ignore(pos, expected))
+        {
+            if (!parseElementsWithoutOptions(pos, expected, elements))
+                return false;
+
+            if (!ParserToken(TokenType::ClosingRoundBracket).ignore(pos, expected))
+                return false;
+        }
+
+        /// If no elements were specified it will grant all available for user grants.
+        /// Using `.size() == 0` because `.empty()` is overridden and returns true for NONE elements.
```

Comment:
The logic with checking `elements.size() == 0` seems too complicated logic, you could just do it if `ParserToken(TokenType::OpeningRoundBracket).ignore(pos, expected)` returned false.

Reply:
This needs to handle `CURRENT GRANTS(NONE ON *.*)` because in this case `elements.size == 1` but `elements.empty()` is true.

Reply:
I thought the following two lines are to support just `GRANT CURRENT GRANTS TO user2`.
And it seems you don't need those lines to support `CURRENT GRANTS(NONE ON *.*)`

Reply:
Yes, you are right, my bad

Reply:
Oh, but actually, no. Those lines also handle `GRANT CURRENT GRANTS() TO user2`, which should behave like `GRANT CURRENT GRANTS TO user2`.

Reply:
Actually, do we even want to support `GRANT CURRENT GRANTS() TO user2`?
Especially after these changes https://github.com/ClickHouse/ClickHouse/pull/48262#issuecomment-1505563215